torch/_dynamo/variables/nn_module.py

```diff
@@ -82,8 +82,14 @@ def convert_to_fake(x):
 @contextmanager
 def record_nn_module_stack(module_key: str, source, tx, mod: torch.nn.Module):
     fully_qualified_name = source.name()
+    num_calls = tx.num_calls.get(fully_qualified_name, 0)
     try:
-        tx.nn_module_stack[module_key] = (fully_qualified_name, mod.__class__)
+        tx.nn_module_stack[module_key] = (
+            fully_qualified_name,
+            mod.__class__,
+            num_calls,
```

Comment:
I wonder if instead of adding another field to the nn_module_stack (which should require a schema minor version bump), we could just modify the key to also contain the num_call? like n_1, n_2..

Reply:
You mean `module_key` or `fully_qualified_name`?

As you can see, there are lots of places that use `nn_module_stack`, nervous about changing more places. It could be that `module_key` is safest...do you roughly know how important it is?

Reply:
For schema minor version bump, didn't see any tests fire.

Reply:
I was referring to `module_key`. I'm not too sure how accurate `module_key` needs to be maintained, but skimming the dynamo code, it looks like it's just generated from this [one function](https://www.internalfb.com/code/fbsource/[2d94c2bdacd7]/fbcode/caffe2/torch/_dynamo/output_graph.py?lines=747-760), so it feels like we could just append the `num_call`?

Reply:
Hmm, I'm not very keen on this idea. Unflatten uses the key as a module id to bucket shared modules; but let's say I can have a local decoder for that format. But the onnx/_internal/fx passes still need to be updated, I think, because they extract the key as "module name" in their metadata structure.

OTOH the utils in ao/quantization seem to only look at values. So does internal AIMP stuff. I won't know what else may rely on keys without trying.

So maybe I can put up a parallel PR that puts this stuff into the key just to check, WDYT?

Reply:
> Unflatten uses the key as a module id to bucket shared modules

Wouldn't fully_qualified_name be a better key to bucket? 

> So maybe I can put up a parallel PR that puts this stuff into the key just to check, WDYT?

Sounds good! Thanks! I'm not opposed to adding another field to the value -- just wanted to see if there was another way to do it.

Reply:
> Wouldn't fully_qualified_name be a better key to bucket?

`self.n` and `self.p` can point to the same module instance, in which case I think they get the same key (at least in Dynamo).

Reply:
CI looks green, please review the other PR. Overall that PR makes fewer changes, so I can close this PR if you wish.

Reply:
Recording offline chat with @angelayi here for posterity. We're going with the alt fix because keys seem less load-bearing than values in `nn_module_stack`, so less chance of breaking unknown workflows.